conda-forge channel support

WWGolay opened the pull request:
Hello,

I'm developing a package that tksheet is a dependency of. I want to put it on conda-forge, but that means all my package's dependencies must also be on conda-forge. This PR has the meta.yaml file needed and adds a GH action to publish to conda (see this link for configuring it with a secret: https://github.com/marketplace/actions/publish-conda-forge).

-Will Golay

ragardner reviewed (changes requested):
Thanks for your PR, 

Any idea if tkinter should be a requirement as it does not come installed with all python installations? such as on pop os

WWGolay commented:
I got the sha256 change done. I also added tk as a requirement. I added a few other fields that previously have been requested when adding packages to conda-forge. Also, I'm happy to do the steps from that link to get tksheet on conda-forge if you'd prefer.

ragardner commented:
Thanks a lot,

I am sorry to mess you around a bit but,

I remembered sorry that the reason the hashes might be different is the zipped files are different on pypi here:
https://pypi.org/project/tksheet/#files

compared to github here:
https://github.com/ragardner/tksheet/archive/6.2.3.tar.gz

I think when you use the command pip install tksheet it uses the pypi link e.g.
https://files.pythonhosted.org/packages/02/d7/2bd895e2558f31b66e53346898e42476cc562a74f69b73d1aeb377769b6f/tksheet-6.2.3.tar.gz

because it would have the files generated by `python setup.py sdist` necessary for pip to install a package
e.g. https://docs.python.org/3/distutils/sourcedist.html

I'm not sure which link would be appropriate to use

I have given a bit of thought to who should add it to conda-forge,

I think given your requirement it might be better if you forked tksheet and did it, I might not be around to support it on conda if you need it modified

I am open to suggestions though

WWGolay commented:
No problem. The SHA256 question is also unclear to me... I did try to run `conda build .` in the root of my local fork of the repo, and it was successfully built, so I suppose it's the correct key. 

Yes, I'm happy to take care of getting it on conda-forge—no worries on that front. I'm also doing this for another package that I need support for.
